### PR TITLE
KAFKA-16655: Deflaking ZKMigrationIntegrationTest.testDualWrite

### PR DESCRIPTION
**Pull request by ahuang98**
This test occasionally fails due to stale broker epoch exceptions, which in turn causes allocate producer ids to fail.

Adds retries as stale broker epoch is a retriable issue, and fixes sendAllocateProducerIds returning 0 as the producerIdStart in error cases

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/integration/kafka/zk/ZkMigrationIntegrationTest.scala [hunk 1]
```diff
@@ -612,7 +613,12 @@ class ZkMigrationIntegrationTest {
       val readyFuture = kraftCluster.controllers().values().asScala.head.controller.waitForReadyBrokers(3)
 
       // Allocate a block of producer IDs while in ZK mode
-      val nextProducerId = sendAllocateProducerIds(zkCluster.asInstanceOf[ZkClusterInstance]).get(30, TimeUnit.SECONDS)
+      var nextProducerId = -1L
+ 
+      TestUtils.retry(60000) {
+        assertDoesNotThrow((() => nextProducerId = sendAllocateProducerIds(zkCluster.asInstanceOf[ZkClusterInstance]).get(20, TimeUnit.SECONDS)): Executable)
+      }
+      assertEquals(0, nextProducerId)
```
Comment (johnnychhsu) [inline core/src/test/scala/integration/kafka/zk/ZkMigrationIntegrationTest.scala:621]: may I know why do we need this `assertEquals(0, nextProducerId)` here?

**Comment by johnnychhsu** [comment]
thanks for the PR! 
got a question and left a comment.

File: core/src/test/scala/integration/kafka/zk/ZkMigrationIntegrationTest.scala (hunk 1 above)
Comment (ahuang98) [reply core/src/test/scala/integration/kafka/zk/ZkMigrationIntegrationTest.scala:621]: this checks if the nextProducerId is allocated correctly before moving onto the next steps

**Comment by ahuang98** [comment]
^ edited title to kick off another build (last one failed due to unrelated issues)

**Review by cmccabe — APPROVED** [review]
LGTM

**Comment by ahuang98** [comment]
test failures look unrelated

**Pull request merged**